internal/subscriber/core_test.go

```diff
@@ -0,0 +1,364 @@
+				modAckCh:               modAckCh,
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockBrokerStore.EXPECT().GetConsumer(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(arg0 context.Context, arg1 messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
+					if tt.wantErr {
+						return nil, errors.New("Test Error")
+					}
+					return cs, nil
+				})
+			mockMessageBroker.EXPECT().Resume(gomock.Any(), gomock.Any()).AnyTimes().Return(nil)
+			sub, err := tt.fields.core.NewSubscriber(tt.args.ctx, tt.args.subscriberID, tt.args.subscription, tt.args.timeoutInMs,
+				tt.args.maxOutstandingMessages, tt.args.maxOutstandingBytes, tt.args.requestCh, tt.args.ackCh, tt.args.modAckCh)
+			if err == nil {
+				assert.True(t, EqualOnly(sub, tt.expected, []string{"topic", "subscriberID", "requestChan", "ackChan", "modAckChan"}))
+			}
+			if (err != nil && !tt.wantErr) || (err == nil && tt.wantErr) {
+				t.Errorf("NewSubscriber wantErr: %v Error: %v", tt.wantErr, err)
+			}
+		})
+	}
+}
+
+func TestCore_NewSubscriber_WithRetry(t *testing.T) {
+	ctx, cs, store, broker, core, offsetCore, ch := setupCore(t)
+	ackCh := make(chan *AckMessage)
+	modAckCh := make(chan *ModAckMessage)
+	requestCh := make(chan *PullRequest)
+	subscrption := getDummySubscriptionWithRetry()
+	type fields struct {
+		core *Core
+	}
+	type args struct {
+		ctx                    context.Context
+		subscriberID           string
+		subscription           *subscription.Model
+		timeoutInMs            int
+		maxOutstandingMessages int64
+		maxOutstandingBytes    int64
+		requestCh              chan *PullRequest
+		ackCh                  chan *AckMessage
+		modAckCh               chan *ModAckMessage
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		expected    *Subscriber
+		expectedMsg []string
+		wantErr     bool
+	}{
+		{
+			name: "Create new subscriber with retry policy",
+			fields: fields{
+				core: getCore(store, core, offsetCore, ch),
+			},
+			args: args{
+				ctx:                    ctx,
+				subscriberID:           subID,
+				subscription:           &subscrption,
+				timeoutInMs:            1000,
+				maxOutstandingMessages: 1000,
+				maxOutstandingBytes:    1000,
+				requestCh:              requestCh,
+				ackCh:                  ackCh,
+				modAckCh:               modAckCh,
+			},
+			expected: &Subscriber{
+				subscription: &subscrption,
+				topic:        topicName,
+				subscriberID: subID,
+				requestChan:  requestCh,
+				ackChan:      ackCh,
+				modAckChan:   modAckCh,
+			},
+			expectedMsg: []string{"a", "b"},
+			wantErr:     false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			store.EXPECT().GetConsumer(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(arg0 context.Context, arg1 messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
+					if tt.wantErr {
```

Comment:
dynamic struct means?

Reply:
{
		name     string
		fields   fields
		args     args
		expected *Subscriber
		msgs     []string
		wantErr  bool
}

Since there is a single test case, why are we having conditional expect calls..?

Reply:
merged these two test cases so now it is conditional expect is required.